libs/coin-framework/src/api/types.ts

```diff
@@ -321,20 +321,47 @@ type MaybeTxData<TxDataType extends TxData> = TxDataType extends TxDataNotSuppor
 
 export type FeesStrategy = "slow" | "medium" | "fast";
 
+export type StakingOperation =
+  | "delegate"
+  | "undelegate"
+  | "redelegate"
+  | "getStakedBalance"
+  | "getUnstakedBalance";
+
 export type TransactionIntent<
   MemoType extends Memo = MemoNotSupported,
   TxDataType extends TxData = TxDataNotSupported,
 > = {
+  intentType: "transaction" | "staking";
   type: string;
   sender: string;
-  senderPublicKey?: string;
-  expiration?: number;
   recipient: string;
   amount: bigint;
-  useAllAmount?: boolean;
   asset: AssetInfo;
-  sequence?: number;
+  useAllAmount?: boolean;
   feesStrategy?: FeesStrategy;
+  senderPublicKey?: string;
+  sequence?: number;
+  expiration?: number;
+} & MaybeMemo<MemoType> &
+  MaybeTxData<TxDataType>;
+
+export type StakingTransactionIntent<
+  MemoType extends Memo = MemoNotSupported,
+  TxDataType extends TxData = TxDataNotSupported,
+> = TransactionIntent & {
+  intentType: "staking";
+  mode: StakingOperation;
+  valAddress: string;
+  dstValAddress?: string;
+} & MaybeMemo<MemoType> &
+  MaybeTxData<TxDataType>;
+
+export type SendTransactionIntent<
+  MemoType extends Memo = MemoNotSupported,
+  TxDataType extends TxData = TxDataNotSupported,
+> = TransactionIntent & {
+  intentType: "transaction";
```

Comment:
OK, then I suggest `"send"` to keep the type names aligned with the discriminator